library/ssl_tls13_keys.c

```diff
@@ -346,4 +347,112 @@ int mbedtls_ssl_tls1_3_evolve_secret(
     return( ret );
 }
 
+int mbedtls_ssl_tls13_populate_transform( mbedtls_ssl_transform *transform,
+                                          int endpoint,
+                                          int ciphersuite,
+                                          mbedtls_ssl_key_set const *traffic_keys,
+                                          mbedtls_ssl_context *ssl /* DEBUG ONLY */ )
+{
+    int ret;
+    mbedtls_cipher_info_t const *cipher_info;
+    const mbedtls_ssl_ciphersuite_t *ciphersuite_info;
+    unsigned char const *key_enc;
+    unsigned char const *iv_enc;
+    unsigned char const *key_dec;
+    unsigned char const *iv_dec;
+
+#if !defined(MBEDTLS_DEBUG_C)
+    ssl = NULL; /* make sure we don't use it except for those cases */
+    (void) ssl;
+#endif
+
+    ciphersuite_info = mbedtls_ssl_ciphersuite_from_id( ciphersuite );
```

Comment:
Generally, I think this is an assertion rather than a genuine check, so we might want to omit it (we certainly can't include assertions for each and everything, and it's up to us to determine where to draw the line) but seeing that the error code returned from the TLS 1.2 version is not `MBEDTLS_ERR_SSL_INTERNAL_ERROR` makes me question that impression. In any case, it should be the same for uniformity, as you say. I'll change it.

Reply:
Done.